src/yunohost/domain.py

```diff
@@ -206,7 +210,22 @@ def domain_remove(auth, domain, force=False):
                                           m18n.n('domain_uninstall_app_first'))
 
     if auth.remove('virtualdomain=' + domain + ',ou=domains') or force:
-        os.system('rm -rf /etc/yunohost/certs/%s' % domain)
+        command_list = [
+            'rm -rf /etc/yunohost/certs/%s' % domain,
+        ]
+
+        if os.path.exists('/etc/letsencrypt/live/%s' % domain):
+            command_list.extend([
+                'yunohost domain letsencrypt revoke %s' % domain,
+                'rm -rf /etc/letsencrypt/archive/%s /etc/letsencrypt/live/%s' % domain,
+                'rm -f /etc/letsencrypt/renewal/%s.conf' % domain,
+                'rm -f /etc/cron.d/letsencrypt-{domain}' % domain
+            ])
+
+        for command in command_list:
+            if os.system(command) != 0:
+                msignals.display(m18n.n('path_removal_failed', command[7:]),
+                                 'warning')
```

Comment:
All this block should be converted to pure python and I think that instead of storing a list of functions to call, calling them directly will simplify the code and make it more maintainable.

Also "yunohost domain letsencrypt revoke" is a function stored in this python file right now, it would made more sens to call it directly (and make code more obvious) :)